### ISSUE TITLE
Cannot use "$1" inside the title property.

### ISSUE THREAD

**Issue by nficano** [nuxt/nuxt#22363]
### Environment

```- Operating System: Darwin
- Node Version:     v18.14.0
- Nuxt Version:     3.6.2
- Nitro Version:    2.5.2
- Package Manager:  yarn@1.22.19
- Builder:          vite
- User Config:      devtools, css, prismic, sitemap, runtimeConfig, build, components, alias, imports, image, modules, nitro, extends, svgo, unhead, postcss, sourcemap, ssr, gtag
- Runtime Modules:  nuxt-svgo@3.2.0, @nuxtjs/prismic@3.0.1, @formkit/nuxt@0.17.4, nuxt-gtag@0.5.7, @nuxt/image-edge@1.0.0-rc.1-28164352.ad7c010
- Build Modules:    -```

### Reproduction

```vue
// app.vue
...
<script setup>
useHead({
  title: 'XYZ training, certification and compliance for $17.95'
})

// --- or ---

definePageMeta({
  title: "XYZ training, certification and compliance for $17.95",
});
</script>
```

Will output:

```html
<head>
  <title>XYZ training, certification and compliance for 7.95</title>
  ...
```

### Describe the bug

If the title property contains the string `$1`, both characters do not render using the dev server or building for release. 

If you use any number other than 1, (e.g.: $0, $2, $3) the title renders correctly.
If you put a space between the $ and 1, it renders correctly

I tried the following workarounds, but none of them work.

- `XYZ training, certification and compliance for $17.95`
- `XYZ training, certification and compliance for ${"$"}17.95`
- 'XYZ training, certification and compliance for ' + '$' + '17.95'

**Comment by danielroe** [comment]
cc: @harlan-zw

**Comment by harlan-zw** [comment]
Hey @nficano, thanks for the issue.

This is fixed in [Unhead v1.1.33](https://github.com/unjs/unhead/releases/tag/v1.1.33), you can update by doing `nuxi upgrade --force` (or deleting lock file + node_modules and re-installing).

You can confirm you're using this version using `npm why unhead`.

Let me know if you have any problems with that :)

**Comment by nficano** [comment]
@harlan-zw you're a wonderful person. Thank you!
